src/conversion.rs

```diff
@@ -440,10 +441,27 @@ impl<'lua, T: ToLua<'lua>> ToLua<'lua> for Vec<T> {
     }
 }
 
-impl<'lua, T: FromLua<'lua>> FromLua<'lua> for Vec<T> {
-    fn from_lua(value: Value<'lua>, _: Context<'lua>) -> Result<Self> {
+impl<'lua, T: FromLuaMulti<'lua>> FromLua<'lua> for Vec<T> {
+    fn from_lua(value: Value<'lua>, lua: Context<'lua>) -> Result<Self> {
         if let Value::Table(table) = value {
-            table.sequence_values().collect()
+            let mut values: MultiValue = MultiValue::from_vec(
+                table
+                    .sequence_values::<Value>()
+                    .filter_map(Result::ok) // Every value in Table is always a Value
+                    .collect(),
+            );
+            let mut result = Vec::new();
+            while values.len() > 0 {
+                let mut consumed = 0;
+                match T::from_counted_multi(values.clone(), lua, &mut consumed) {
```

Comment:
The main drawback with that approach is that implementations have to manually juggle values (revert them when testing fails). `Fallible` could do a clone and revert it on error but it complicates the implementation for most other structures.

I tried adding a wrapper as well at some point but it required adding lifetimes to all the traits/functions that deal with `FromLuaMulti` conversion which felt like it complicates code too much. Though I do think an iterator with "revert last argument" function and "list skipped types" function would be ideal.

Reply:
I agree that adding extra lifetimes to traits would be awkward, and make updates from previous versions difficult.
But I think the quadratic runtime for long (even non-multi-value) argument lists is also a problem.
That quadratic issue is only because multiple arguments can have a crack a (nearly) the same list, and wouldn't be a problem with the old "take all" version.  Maybe the new counted method can take a different type which does behave as an iterator.  Perhaps with a "peek next" rather than "revert pop"?